app/controllers/topics_controller.rb

```diff
@@ -76,6 +76,14 @@ def respond_for_topic_publication(publication, opts = {})
     end
   end
 
+  def next_position_in_list(topic)
+    highest_position_in_list(topic) + 1
+  end
+
+  def highest_position_in_list(topic)
+    topic.topic_sections.map(&:position).max || 0
```

Comment:
Didyaknow `topic.topic_sections.max_by(&:position)` would do the same? No need to change of course. They're virtually identical.